cpp/ql/test/library-tests/controlflow/guards-ir/ASTGuardsCompare.expected

```diff
@@ -14,6 +14,10 @@
 | 17 | y < 1+1 when ... > ... is false |
 | 17 | y >= 1+1 when ... && ... is true |
 | 17 | y >= 1+1 when ... > ... is true |
+| 18 | call to get != call to get+0 when (bool)... is true |
+| 18 | call to get != call to get+0 when call to get is true |
+| 18 | call to get == call to get+0 when (bool)... is false |
+| 18 | call to get == call to get+0 when call to get is false |
```

Comment:
The conversion to bool here is the same result as above. The equality constraints are new results of comparesEq, which I believe are correct, but may not be particularly useful.